app/models/cloud_volume.rb

```diff
@@ -161,6 +168,39 @@ def raw_delete_volume
     raise NotImplementedError, _("raw_delete_volume must be implemented in a subclass")
   end
 
+  # ========================================== safe-delete ==========================================
+
+  def safe_delete_volume_queue(userid)
+    task_opts = {
+      :action => "Safe deleting Cloud Volume for user #{userid}",
+      :userid => userid
+    }
+
+    queue_opts = {
+      :class_name  => self.class.name,
+      :method_name => 'safe_delete_volume',
+      :instance_id => id,
+      :role        => 'ems_operations',
+      :queue_name  => ext_management_system.queue_name_for_ems_operations,
+      :zone        => ext_management_system.my_zone,
+      :args        => []
+    }
+
+    MiqTask.generic_action_with_callback(task_opts, queue_opts)
+  end
+
+  def safe_delete_volume
+    raw_safe_delete_volume
+  end
+
+  def raw_safe_delete_volume
+    raise NotImplementedError, _("raw_safe_delete_volume must be implemented in a subclass")
+  end
+
+  def validate_safe_delete_volume
```

Comment:
Do you need `supports?(:safe_delete)` and `validate_safe_delete_volume` ?

I don't see this used by your UI or API PR

Reply:
Ah, I thought the validate maybe used behind the scenes somewhere.

Reply:
:+1: older UI code used to use this but anything new should be using the supports feature that you added above